shell/test/specs/demo-tests.js

```diff
@@ -304,20 +304,27 @@ function allSuggestions() {
   waitForStillness();
   openSuggestionDrawer();
 
+  console.log(`click: allSuggestions`);
+  clickMagnifier();
```

Comment:
So the restaurant demo *is* calling `allSuggestions` at its beginning, which in theory does the click and would open the suggestions. But I believe you when you say that they're not showing up. So I would then assume this PR fixes things because it causes another click, later, when perhaps some processing that takes a while and prevents suggestions from showing up is done.

Does that mesh with your expectation, or what do you think is happening?

I'd support landing this if so but we may want to look more closely at what's taking so long or otherwise holding up suggestions from the first click.

Reply:
I've simplified this patch and moved the browser.waitUntil to the allSuggestions().

It seems that dancing dots are no longer dancing, this might be the root cause of why we try looking at suggestions prematurely though.